lib/subscription-client.js

```diff
@@ -204,6 +206,9 @@ class SubscriptionClient {
         break
       case GQL_CONNECTION_ERROR:
         this.close(this.tryReconnect, false)
+        if (this.failedConnectionCallback) {
+          this.failedConnectionCallback(data.payload)
```

Comment:
I would add an await here.

Reply:
Done, I've not handled the case were `failedConnectionCallback` throws because we don't have the fastify instance to log the error in the `SubscriptionClient`, this would require a modification on `buildServiceMap`